File: sort/inssort.cpp
// 2020 3 20
// Insertion sort implementation using C++

#include <iostream>

using namespace std;

void print(int lib[], int n){
    for(int i=0; i<n; i++) cout << lib[i] << " ";
}

void inssort(int lib[], int n){
    int j=0;
    for(int i=0; i<n; i++){
        j = i+1;
        while(lib[j]<lib[i]){
            swap(lib[j],lib[i]);
            --j;
            --i;
            if(i<0) break;
        }
    }
}

int main(){
    int lib[] = {9,25,40,36,1,8};
    int n = sizeof(lib)/sizeof(lib[0]);
    inssort(lib,n);
    print(lib,n);
    return 0;
}

File: sort/inssort.py
# 2020 3 20
# Insertion sort implementation using Python

lib = [9,25,40,36,1,8]

def inssort(lib):
    for i in range(0,len(lib)-1):
        j = i+1
        while(lib[j]<lib[i]):
            lib[j],lib[i] = lib[i],lib[j]
            j-=1
            i-=1
            if(i<0): break

inssort(lib)
print(lib)